### ISSUE TITLE
Issues with copy-paste from keyboard on OS X El Capitan

### ISSUE THREAD

**Issue by brettneese** [zeit/hyper#484]
Having issues with paste using CMD-V on my OSX machine. Hyperterm flat-out refuses to acknowledge my keyboard command.

Other key commands, such as CMD-+ and CMD-- work just fine, however, as do the menu commands. I thought it was just not implemented yet, but I haven't seen any other issues with this reported on OSX, so it looks like a bug.

The only variable on my machine I can think of is that I am using http://pasteapp.me/helper, however, killing Paste altogether does not change anything.
## 

HyperTerm 0.7.1
Electron 1.2.8
darwin x64 15.4.0

**Comment by matheuss** [comment]
If you select some text on HyperTerm you can't `cmd + c` it?

**Comment by brettneese** [comment]
OK, cmd-c is working, but I don't think it was in 0.7. cmd-v is still not working.

**Comment by jamesfzhang** [comment]
I've been able to reproduce this bug by trying to paste something twice. The first paste works, but the second paste yields nothing.

**Comment by aeromusek** [comment]
Like @jamesfzhang, I see this when trying to paste something twice. It happens reliably when inside an SSH connection. Killing the connection and reconnecting solves the problem.

**Comment by matheuss** [comment]
I still can't reproduce 😰 (Tried inside `ssh`, same result)
![cant-reproduce](https://cloud.githubusercontent.com/assets/4721750/17374598/c417f1ae-5983-11e6-9675-4d2be8310475.gif)

**Comment by brettneese** [comment]
Yeah, this is super weird. Anything I can do on my end to help debug?

**Comment by matheuss** [comment]
The console says anything? (`Cmd + Option + I`)

**Comment by brettneese** [comment]
![image](https://cloud.githubusercontent.com/assets/504279/17377968/c5c64002-5981-11e6-82df-0acbcb03e8c7.png)

Only this. Interestingly, I can paste into the console.

**Comment by jamesfzhang** [comment]
Nothing from console, but if I refresh (`Cmd + R`), I can paste multiple times until I press enter, after which I can only paste once.

![copypasta](https://cloud.githubusercontent.com/assets/2057670/17378237/fde7d712-5971-11e6-8180-79a671d69e11.gif)

**Comment by chabou** [comment]
Paste only once ? Even after adding a space after the first past ?

**Comment by chabou** [comment]
Maybe the same origin than #479

**Comment by rauchg** [comment]
I'm assuming this has something to do with focus, like the paste being captured by electron vs the renderer process. Worth investigating in that direction. Potentially capturing the pasteboard from electron and passing it down for example as a `paste` rpc event.

**Comment by matheuss** [comment]
Closing in favor of #479

**Comment by brettneese** [comment]
:(

I don't think these issues are actually related, I can't paste even the first time.

**Comment by matheuss** [comment]
Ops, sorry!

**Comment by matheuss** [comment]
I didn't notice that you *can't* paste at all! 😱

**Comment by brettneese** [comment]
It's ok, I should've clarified sooner, I just let sort of let the other people comment. It's possible they're related, I'm just not sure yet. Very odd that I'm the only one experiencing this, my guess is there's something weird with my machine but if there is I'd like to figure it out. 😮

**Comment by matheuss** [comment]
What `shell` you are using? Any theme, plugin, extension or something that you are using on HyperTerm/your shell? Also, anything *different* on your `.zshrc || .bashrc || .bash_profile`?

**Comment by curioussavage** [comment]
on Version 0.8.1 in neovim I cannot copy/paste or select text

**Comment by atdrago** [comment]
@brettneese You're not the only one. This happens to me constantly since updating to Sierra. I'll copy a bit of text, then go to paste it and nothing will happen. Open Terminal.app, paste the text there and it works fine. Back to Hyper.app, nothing. I see the "Edit" menu light up, but no paste.

One thing that's somewhat interesting: If I open a new tab, paste works fine. Back to the first tab, paste still doesn't work. Open Developer Tools, paste works again (inside the term). Seeing the same 'Unknown OSC Code: "1"' warning. Also, a "dead key" warning. Not sure what that's about...

![screen shot 2016-10-10 at 9 55 17 am](https://cloud.githubusercontent.com/assets/144225/19244127/b3f53ea4-8ecf-11e6-88ed-ab04c8a6b08a.png)

**Comment by atdrago** [comment]
@matheuss Asked for these:

**.zshrc** (uncommented lines only)
```
export ZSH=/Users/adamdrago/.oh-my-zsh

ZSH_THEME="robbyrussell"

plugins=(git)

source $ZSH/oh-my-zsh.sh
```

**.bash_profile**
```
export NVM_DIR="/Users/adamdrago/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"  # This loads nvm
export PS1="[\t] \W \e[0;32m\$\e[m "
```

**Comment by ekmartin** [comment]
The vim paste issues are because of https://github.com/zeit/hyper/issues/773.

**Comment by atdrago** [comment]
Some more info:

It seems like it's a focus issue like @rauchg said. Some test cases:

**Paste fails**
1. Type `git diff`
2. `CMD+V`

**Paste fails**
1. Type anything
2. Delete what you typed
2. `CMD+V`

**Paste succeeds**
1. `CMD+V` on a clear line

**Paste succeeds**
1. Type `git diff`
2. Highlight some other text in the term
3. `CMD+V`

**Paste succeeds**
1. Type `git diff`
2. Click the line I'm typing on
3. `CMD+V`

In other words, if I type at all on a line, I can't paste anymore on that line until I somehow retrigger focus. I'm hoping this is helpful. Not trying to spam this issue. Please let me know if there's anything else I can do. :)

**Comment by jtphelan** [comment]
Same issue as @atdrago. It's like hyper loses focus even though I can still type. If I click on the app a couple times and then try and paste it works.

**Comment by atdrago** [comment]
Not yet fixed in 0.8.2

**Comment by rezozo** [comment]
This also happens to me all the time. I'm on 0.8.3

**Comment by tylerflint** [comment]
Same here. It is a constant headache. As soon as I click in the terminal window the paste works.

**Comment by hwkr** [comment]
Same problem, super annoying. I find sometimes waiting a couple seconds also makes it work.

**Comment by matheuss** [comment]
Yo y'all, could you test on latest master? I might have introduced a fix via #1006 👌

**Comment by jwhitley** [comment]
@matheuss  Paste appears to work great now; I haven't been able to get a paste failure yet.

I hit clipboard testing using these methods, all passed:
 * Some ad-hoc ⌘-C/V testing
 * neovim `*` register testing
 * pbcopy/pbpaste
 * @atdrago's fail cases above

I'd encourage others to give it a try as well, just in case I've missed anything.

Thanks!  This was a big blocker for my getting into hyper.  💯

**Comment by aeromusek** [comment]
@matheuss @jwhitley ditto. Fixed for me too!

**Comment by matheuss** [comment]
Closing since it was fixed 🙌

**Comment by SteveAzz** [comment]
How is this done in Linux? `ctrl+c` doesn't works neither does `ctrl+shift+c`

**Comment by ppot** [comment]
@SteveAzz There is an issue for linux. regarding those keystrokes
